fix(shield): grouped expressions not being handled at all

pitasi opened the pull request:
We were completely ignoring grouped expressions (e.g. expressions wrapped between parentheses such as `(true || false)`).

This PR adds proper support and tests for this cases.

pitasi committed:
fix(shield): handle InfixExpression in ast.Stringify

An entire type of expression was missing in this Stringify method.

I also added a check in case `exp` is nil, so that it's a lot more
evident during debugging if something is missing (e.g. if a field from
an expression is `nil`).

pitasi committed:
feat(shield): add missing handler for grouped expressions

This was a big missing piece for the shield parser, it was not handling
grouped expressions (i.e. expressions wrapped between (...)) at all.

pitasi committed:
test(shield): increase Parser test coverage

pitasi committed:
chore(changelog): fix shield grouped expression

mn13 commented on shield/internal/parser/parser_test.go:
```diff
@@ -96,3 +96,49 @@ func TestBooleanOperations(t *testing.T) {
+		{
+			`true || (true && false)`,
+			"(true || (true && false))",
+		},
```
Hey @Pitasi I suggest to add case with built-in function call, something like this

```
{
  `((true || (address) || ((any(2, true) || false) && true)))`,
  "((true || address) || ((any(2, true) || false) && true))",
}
```
It will require a little change is Stringify, I believe it can be done as Stringify is not used anywhere except tests
```
case *Expression_CallExpression:
		return fmt.Sprintf("%s(%s)", n.CallExpression.Function.Value, stringifyExpressions(n.CallExpression.Arguments))
```

pitasi replied:
Good catch, thanks, it was another bug inside the Stringify function. I have fixed it now in [d91da01](https://github.com/warden-protocol/wardenprotocol/pull/225/commits/d91da0158ce4a068648bc1254543dcd48ee892e5) and added the new test case.

pitasi committed:
fix(shield): Stringify function name correctly

We were serializing a struct (n.CallExpression.Function) like it was
just the name of the string, but it wasn't.
